### PR TITLE
move to hardhat-circom

### PR DESCRIPTION
**Pull request by jacobrosenthal**
Demo of how hardhat-circom might look

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -5,12 +5,14 @@
   "license": "MIT",
   "dependencies": {},
   "devDependencies": {
-    "circom": "^0.5.34",
-    "circomlib": "^0.4.1",
-    "dotenv": "^8.2.0",
-    "snarkjs": "^0.3.51"
+    "hardhat": "^2.2.0",
+    "@darkforest/hardhat-circom": "../dark-forest/darkforest/packages/hardhat-circom"
```
Comment (HAOYUatHZ) [inline package.json:9]: Can we avoid relative path here? or maybe add some instruction for prerequisite installation in README?

Comment (jacobrosenthal) [reply package.json:9]: Yes. its relative to an optional base path (or made absolute to the hardhat root dir)
Sorry I dont have it published yet, but this is what a fully overriden option set might look like (I was trying to keep this repo simple looking)


```js
module.exports = {
  circom: {
    verifierTemplatePath: './contracts/Verifier.sol.template',
    verifierOutName: 'Verifier.sol', // Output path is hardcoded to hardhat artifact directory
    circuitInputBasePath: './circuits/',
    circuitOutputBasePath: '../client/public/',
    circuits: [
      {
        name: 'init',
        circuit: 'init/circuit.circom',
        input: 'init/input.json',
        ptau: 'init/pot15_final.ptau',
        wasm: 'circuits/init/circuit.wasm',
        zkey: 'init.zkey',
        beacon: '0102030405060708090a0b0c0d0e0f101112131415161718191a1b1c1d1e1f',
      },
    ],
  },
};
```

**Comment by HAOYUatHZ** [comment]
Usually it will be good if we can avoid upload such binary files as hash.wasm and hash.zkey. But I guess if we really wanna use hardhat-circom, then for convenience we have to upload them.

File: hardhat.config.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
+require('hardhat-circom');
+
+/**
+ * @type import('hardhat/config').HardhatUserConfig
+ */
+module.exports = {
+  solidity: "0.6.7",
+  circom: {
+    ptau: "hash.ptau",
```
Comment (phated) [inline hardhat.config.js:9]: Should we update to the hermez pot15 final ptau? (and keep their downloaded name)

File: package.json [hunk 3]
```diff
@@ -5,12 +5,14 @@
   "license": "MIT",
   "dependencies": {},
   "devDependencies": {
-    "circom": "^0.5.34",
-    "circomlib": "^0.4.1",
-    "dotenv": "^8.2.0",
-    "snarkjs": "^0.3.51"
+    "hardhat": "^2.2.0",
+    "hardhat-circom": "1.0.0"
```
Comment (phated) [inline package.json:9]:
```suggestion
    "hardhat-circom": "^1.0.0"
```

File: package.json (hunk 1 above)
Comment (jacobrosenthal) [reply package.json]: Now published as 1.0.0
https://github.com/projectsophon/hardhat-circom/releases/tag/v1.0.0

File: hardhat.config.js (hunk 2 above)
Comment (jacobrosenthal) [reply hardhat.config.js:9]: Yes

**Review by HAOYUatHZ — APPROVED** [review]

**Comment by jacobrosenthal** [comment]
Needs better docs. Ideally a tutorialized version of the snarkjs and hardhat-circom readmes. That would supercede #1
not sure Ill get to that here

**Review by phated — APPROVED** [review]
Someone probably wants to update the ptau to one of the latest hermez ones from https://www.dropbox.com/sh/mn47gnepqu88mzl/AACaJkBU7mmCq8uU8ml0-0fma?dl=0 but let's get this merged anyway.

**Comment by jacobrosenthal** [comment]
Closes #2 sneakily

**Pull request merged**